### ISSUE TITLE
How to use frida with C#

### ISSUE THREAD

**Issue by salvadorc17** [frida/frida-clr#5]
Hi i want to use frida with C# to create application that inject code into a game, but i was not able to run this app, it ask for missing assembly and i do have Frida.dll

**Comment by oleavr** [comment]
Hi,

Is Frida.dll for the correct architecture? For example if you use the default of AnyCpu on a 64-bit system but have the 32-bit Frida.dll.

Cheers!

**Comment by salvadorc17** [comment]
Oh yes, that was seems  the problem, but now i got BadImageFormat exception becuase cant start ddeviceManager = new DeviceManager(Dispatcher)

**Comment by salvadorc17** [comment]
Also any other dll is needed (gumpp.dll ??))

**Comment by oleavr** [comment]
> Oh yes, that was seems the problem, but now i got BadImageFormat exception becuase cant start
> ddeviceManager = new DeviceManager(Dispatcher)

Which Windows version are you trying this on? It should work fine on Windows 10, but I've heard of an issue on 7 (and possibly 8), which @riverar is looking into. (Seems to be a bug in Microsoft's CRT related to Thread-Local Storage callbacks in managed DLLs.)

**Comment by oleavr** [comment]
> Also any other dll is needed (gumpp.dll ??))

No, none.

**Comment by salvadorc17** [comment]
Yes, im on Windows 7 64 bits, so no way to use it right now becuase that??

**Comment by riverar** [comment]
Sadly, the C# bindings are affected on Windows 7. You can either upgrade to a more modern OS or use the python/node bindings for now.

**Comment by Iquis** [comment]
No news about a fix for Windows7?

**Comment by D-4-N** [comment]
@oleavr any updates on this?

**Comment by oleavr** [comment]
@D-4-N Still waiting for someone to contribute their time to investigate. Might be fixed, though, now that we build GLib for each type of C-runtime, but I haven't checked.

**Comment by riverar** [comment]
@D-4-N I'd recommend opening a new issue to cover your case. Frida has changed significantly since this issue was opened years ago.

**Comment by D-4-N** [comment]
@oleavr Looking forward to it
@riverar Will do, thank you
